docker_runner.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -x
+
+if [ "$HOME" = "/home/runner" ]; then
+  my_path="."
+else
+  my_path="/vagrant"
+fi
+
+docker build -t beyond/web_app $my_path
+docker run -t -d --rm --name beyond beyond/web_app
```

Comment:
Is there a reason not to use the vagrant docker provisioner to run the container?

Reply:
Yes, because then the docker container would be running only when we run the Vagrant up.
On the CI flow it wont run if i would use the Vagrant docker provisioner.
I wanted to use the container both, in CI and regular Vagrant flow.
Thats why i took it out to external shell script, so i could invoke it both, from the Vagrant file AND the main.yml (from the github flow).

Reply:
This is a valid point. docker-compose is a better suited technology then bash.

Reply:
I am not very familiar with docker-compose, and how it can replace invocation of this shell script from two different places. But i will try to figure this out...

Reply:
I read a bit about docker-compose, to get a clue about what it is and what it is used for.
But still i dont understand how this can help us with the current duplication we have in invocation of 'docker run' (1- in docker_runner.sh -> for the 'Vagrant up' flow    2- in main.yml  -> for the CI flow.
We still need to invoke it from two sources if we wish to both - test in in the CI AND use it with the Vagrant flow.

I think for now we should review it again and consider merging it, since this task didnt define the usage of docker-compose in the first place. So I implemented it on the best side, but as i understood to make this possible.

In the future we could consider converting the flow to use docker compose.